packages/core/strapi/lib/types/utils/expression.d.ts

```diff
@@ -14,6 +14,8 @@ export type IsFalse<TValue> = [TValue] extends [False] ? True : False;
 
 export type Extends<TLeft, TRight> = [TLeft] extends [TRight] ? True : False;
 
+export type NotExtends<TLeft, TRight> = Not<Extends<TLeft, TRight>>;
```

Comment:
I think I suggested `NotExtends` this morning, but now I'm leaning towards `DoesNotExtend` because it's more clear.  I'm fine either way though, happy to hear what anyone else thinks.

Reply:
cc for opinion if you have one: @Marc-Roig @alexandrebodin @christiancp100 @Bassel17 @joshuaellis

Reply:
I think bot options clearly express the intention of negating an expression.

DoesNotExtends sounds more like natural English to me so maybe I would pick that one too.  Still, I wouldn't mind having one or another. Just that I would stick to one of them for cases like this

Reply:
> DoesNotExtend sounds more like natural English to me, so maybe I would pick that one too.

I agree with this if i'm being picky, `NotExtends` just sounds clunky

Reply:
Since we're using `IsNever` `IsNotNever`, I would go with `Extends` `DoesNotExtends` to be consistent. Thanks for your input everyone :100: